softwareComponents/configuration/include/configuration/serialization.hpp

```diff
@@ -31,14 +31,14 @@ namespace rofi::configuration::serialization {
             assert( false && "String does not represent a component type" );
     }
 
-    template< typename M >
-    M fromJSON( const nlohmann::json& j, ModuleId id );
+    template< typename M > requires std::derived_from< M, Module >
+    M moduleFromJSON( const nlohmann::json& j, ModuleId id );
 
-    template< typename M >
-    M fromJSON( const nlohmann::json& j ) {
+    template< typename M > requires std::derived_from< M, Module >
+    M moduleFromJSON( const nlohmann::json& j ) {
         assert( j.is_object() && ++j.begin() == j.end() && "json is not a single object" );
```

Comment:
This is undefined when there is no object.
Also I would like it to throw an exception, since it's essentialy reading the user input (so you want to catch that even in production code).

PS. And `stoi` can already throw.

Reply:
Just to be clear: the UB occurs when there are no children; if so, `++j.begin()` is undefined.